tests/frontend/test_decompositions.py

```diff
@@ -288,6 +288,59 @@ def test_random_unitary_phase_end(self, tol):
 
         assert np.allclose(U, qrec, atol=tol, rtol=0)
 
+    def test_identity_MZ(self, tol):
+        """This test checks the rectangular decomposition for an identity unitary.
+
+        An identity unitary is decomposed via the rectangular decomposition of
+        Clements et al. and the resulting beamsplitters are multiplied together.
+        Test passes if the product matches identity.
+        """
+        # TODO: this test currently uses the T and Ti functions used to compute
+        # Clements as the comparison. Probably should be changed.
+        n = 20
+        U = np.identity(n)
+
+        tilist, diags, tlist = dec.rectangular_MZ(U)
+
+        qrec = np.identity(n)
+
+        for i in tilist:
+            qrec = dec.mach_zehnder(*i) @ qrec
+
+        qrec = np.diag(diags) @ qrec
+
+        for i in reversed(tlist):
+            qrec = dec.mach_zehnder_inv(*i) @ qrec
+
+        assert np.allclose(U, qrec, atol=tol, rtol=0)
+
+    def test_random_unitary_MZ(self, tol):
+        """This test checks the rectangular decomposition for a random unitary.
+
+        A random unitary is drawn from the Haar measure, then is decomposed via
+        the rectangular decomposition of Clements et al., and the resulting
+        beamsplitters are multiplied together. Test passes if the product
+        matches the drawn unitary.
+        """
+        # TODO: this test currently uses the T and Ti functions used to compute
```

Comment:
Is this meant to fixed in this PR?

Reply:
Oops, this was copied from other tests. I don't really know what this todo means, so removed this from the docstrings of `test_random_unitary_MZ` and `test_identity_MZ`